### ISSUE TITLE
Upscayl does not use GPU ID entered

### ISSUE THREAD

**Issue by gaiking-uk** [upscayl/upscayl#337]
### Checklist

- [X] I have checked that this issue isn't addressed in [the FAQ](https://github.com/upscayl/upscayl#faq-).
- [X] I have checked that this issue isn't addressed in any other issue or pull request.

### Describe the Bug

I have 2 GPUs in my machine...
1. Basic built-in graphics chip -- Intel UHD Graphics 630 (perf. score: 1,260)
2. Dedicated GPU -- Nvidia GTX 1660Ti (perf. score 8686)

Given my dedicated GPU is about 7x more powerful than my built-in graphics chip, I naturally want Upscayl to use this one -- I followed the instructions to get the GPU IDs, which were...
- `0` = The built-in 630 chip
- `1` = GTX 1660Ti card

I entered `1` as the GPU ID and Upscayled some more images, but this time the upscayl was noticeably slower. I checked my system resources and found that Upscayl was using `GPU 0` (built-in graphics) instead.

I reversed the setting (entering `0` as the GPU ID, just in case) but Upscayl continued to run slowly, and checking it's usage again, it was still only using the built-in graphics, and my graphics card was idle.

I deleted the GPU ID (` `) and this time Upscayl ran much faster and (as you might expect) checking the system resource usage confirmed that this time it was ignoring the built-in graphics and using my much more powerful graphics card.

> PS - I realise as there is a specific setting for this, this is something that has been tested and is working, however I can't explain that happened other than Upscayl just seeming to ignore the GPU ID

<details>
<summary>Logs</summary>
<pre>

🚧 FOLDER_UPSCAYL_PROGRESS:  [0 Intel(R) UHD Graphics 630]  queueC=0[1]  queueG=0[1]  queueT=0[1]
[0 Intel(R) UHD Graphics 630]  bugsbn1=0  bugbilz=83  bugcopc=0  bugihfa=0
[0 Intel(R) UHD Graphics 630]  fp16-p/s/a=1/1/1  int8-p/s/a=1/1/1
[0 Intel(R) UHD Graphics 630]  subgroup=32  basic=1  vote=1  ballot=1  shuffle=1
🚧 FOLDER_UPSCAYL_PROGRESS:  [1 NVIDIA GeForce GTX 1660 Ti with Max-Q Design]  queueC=2[8]  queueG=0[16]  queueT=1[2]
[1 NVIDIA GeForce GTX 1660 Ti with Max-Q Design]  bugsbn1=0  bugbilz=0  bugcopc=0  bugihfa=0
[1 NVIDIA GeForce GTX 1660 Ti with Max-Q Design]  fp16-p/s/a=1/1/1  int8-p/s/a=1/1/1
[1 NVIDIA GeForce GTX 1660 Ti with Max-Q Design]  subgroup=32  basic=1  vote=1  ballot=1  shuffle=1
</pre>
</details>

### To Reproduce

1. Run Upscayl on a machine with multiple GPU devices and enter a GPU ID
2. Upscayl does not use the value entered and instead defaults to the first GPU detected(?)

### Expected Behavior

Upscayl uses the CPU ID entered

### Version/Commit Hash

2.5.1

### OS Specifics

Windows 11 22H2

**Comment by NayamAmarshe** [comment]
It seems that Real-ESRGAN backend somehow doesn't really use GPU-ID correctly 100% of the time.

I'll have to reproduce this issue to confirm.

**Comment by gaiking-uk** [comment]
@NayamAmarshe -- thanks... I don't know if it is something to do with the built-in graphics over-riding something or having priority?

As in, maybe with 2x dedicated GPU cards it works fine - but with 1x built-in (directly into the motherboard) and 1x GPU card, it doesn't read/use the setting correctly

( just a guess )

**Comment by NayamAmarshe** [comment]
> @NayamAmarshe -- thanks... I don't know if it is something to do with the built-in graphics over-riding something or having priority?
> 
> As in, maybe with 2x dedicated GPU cards it works fine - but with 1x built-in (directly into the motherboard) and 1x GPU card, it doesn't read/use the setting correctly
> 
> ( just a guess )

Could you try launching Upscayl in performance mode? (Under Windows advanced display settings)

**Comment by gaiking-uk** [comment]
> Could you try launching Upscayl in performance mode? (Under Windows advanced display settings)

Hi @NayamAmarshe -- Apologies for the delay, I was on holiday and missed your reply... Good news though -- **this worked!** 🎉 great thinking and thanks for your help on this! 

***

#### Additional info
In case it is useful for yourself or other readers, based on my testing...

When `upscayl.exe` is **not set** in performance mode (*not listed* at all), or presumably when set to "Let Windows decide"...
❌ If a GPU ID is entered (in `upscayl.exe` settings), Windows will override this and use the 'power saving' GPU
⚠️ This may significantly slow performance
✅ Leaving the GPU ID empty will let `upscayl.exe` auto-pick the best (most powerful) GPU

When `upscayl.exe` is set to **performance mode**, it work as expected (much better)...
✅ If a GPU ID is entered (in `upscayl.exe` settings), upscayl will use the GPU listed
✅ Leaving the GPU ID empty will let `upscayl.exe` auto-pick the best (most powerful) GPU

**Comment by gaiking-uk** [comment]
PS -- Just before closing the issue, one thing I did note when I was doing my benchmarks to test this issue out (which relates to [the other ticket I raised about the percentage complete indicator](https://github.com/upscayl/upscayl/issues/336))....
1. The percentage complete is not helpful really at all when doing a batch upscayl, the only way I could really check the % complete was to look at the output folder
2. When `upscayl.exe` was running in the foreground it increased the load on my secondary GPU by 10%, minimising `upscayl.exe` removed this 10% GPU drain... Again, my knowledge is admittedly limited but it seems like generating the swirling dots and complete percentage takes a non-tiny amount of GPU effort

Given the two points above, I understand the reasons why the % complete is wrong in batch/folder mode but could I politely re-raise my request to address this somehow -- e.g. even if it means disabling the % complete for batches?
